src/components/cl/basic/cl_basic_team.c

```diff
@@ -12,33 +12,31 @@ UCC_CLASS_INIT_FUNC(ucc_cl_basic_team_t, ucc_base_context_t *cl_context,
 {
     ucc_cl_basic_context_t *ctx =
         ucc_derived_of(cl_context, ucc_cl_basic_context_t);
-    int                     nteams = 0;
+    int                     n_teams = 1, i;
     ucc_status_t            status;
 
     self->tl_ucp_team  = NULL;
     self->tl_nccl_team = NULL;
     ucc_assert(ctx->tl_ucp_ctx != NULL);
     UCC_CLASS_CALL_SUPER_INIT(ucc_cl_team_t, &ctx->super);
-    status = ucc_team_create_multiple_req_alloc(&self->team_create_req,
-                                                UCC_CL_BASIC_NUM_TLS);
+    status = ucc_team_multiple_req_alloc(&self->team_create_req,
+                                         UCC_CL_BASIC_NUM_TLS);
     if (UCC_OK != status) {
         cl_error(cl_context->lib, "failed to allocate team req multiple");
         return status;
     }
-    self->team_create_req->descs[nteams].ctx = ctx->tl_ucp_ctx;
-    memcpy(&self->team_create_req->descs[nteams].param, params,
-           sizeof(ucc_base_team_params_t));
-    self->team_create_req->descs[nteams].param.scope    = UCC_CL_BASIC;
-    self->team_create_req->descs[nteams].param.scope_id = 0;
-    nteams += 1;
-    if (ctx->tl_nccl_ctx != NULL) {
-        self->team_create_req->descs[nteams].ctx = ctx->tl_nccl_ctx;
-        memcpy(&self->team_create_req->descs[nteams].param,
-               &self->team_create_req->descs[nteams-1].param,
+    self->team_create_req->descs[0].ctx = ctx->tl_ucp_ctx;
+    if (self->tl_nccl_team) {
```

Comment:
I think it should be if (self->tl_nccl_ctx)

Reply:
👍  fixed